### DISCUSSION TITLE
rate limiting

### DISCUSSION THREAD

**Discussion by Jooaomar** [honojs/hono#2556] [Q&A] [answered]
How can I impose rate limiting on routes?

**Comment by NicoPlyley** [comment, marked as answer]
Someone in the community has made a package for it that can also work in severless environments https://github.com/rhinobase/hono-rate-limiter

**Reply by Jooaomar** [reply]
Thanks!

**Comment by yasuaki640** [comment]
If your app is running on Cloudflare Workers, [rate limiting is a built-in feature of Workers.](https://developers.cloudflare.com/workers/runtime-apis/bindings/rate-limit/)
(However, it still appears to be a beta feature).

**Reply by Jooaomar** [reply]
Thanks!
